app/views/provider/admin/keys/edit.html.erb

```diff
@@ -1,14 +1,24 @@
-<h2>Set Custom User Key</h2>
+<div class="pf-c-card__title">
+  &nbsp;
+</div>
+<div class="pf-c-card__body">
+  <%= semantic_form_for @cinstance, url: provider_admin_application_key_path(@cinstance, @cinstance.user_key),
+                                    remote: true,
+                                    html: { class: 'pf-c-form pf-m-limit-width' }  do |form| %>
+    <div id="invalid-key-warning" class="pf-c-alert pf-m-danger pf-m-inline" style="display:none;">
+      <div class="pf-c-alert__icon">
+        <i class="fas fa-fw fa-exclamation-circle" aria-hidden="true"></i>
+      </div>
+      <p class="pf-c-alert__title"></p>
+    </div>
 
-<p id="invalid-key-warning" style="display:none;">&nbsp;</p>
+    <%= form.input :user_key, as: :patternfly_input, required: true, input_html: { size: 100, value: nil, required: false } %>
```

Comment:
Hm, this `required: true` and `required: false` thing is quite confusing :upside_down_face:

Reply:
So it is expected to have it like this? Wouldn't one override another?

Reply:
No, it's not expected. One is for formtastic's internal logic, another is for the html tag.